src/graphql/getter.ts

```diff
@@ -133,21 +144,68 @@ export default class GraphQLGetter extends CommandBase {
     return this;
   };
 
-  withNearImage = (args: NearImageArgs) => {
+  private withNearMedia = (args: NearMediaArgs) => {
     if (this.includesNearMediaFilter) {
       throw new Error('cannot use multiple near<Media> filters in a single query');
     }
-
     try {
-      this.nearImageString = new NearImage(args).toString();
+      this.nearMediaString = new NearMedia(args).toString();
+      this.nearMediaType = args.type;
       this.includesNearMediaFilter = true;
     } catch (e: any) {
       this.addError(e.toString());
     }
-
     return this;
   };
 
+  withNearImage = (args: NearImageArgs) => {
+    return this.withNearMedia({
+      ...args,
+      type: NearMediaType.Image,
+      media: args.image ? args.image : 'UNSET',
```

Comment:
why only for `NearImage` with have this check of `image` existance then if not set we set `UNSET` value?

Reply:
It's because of the need for backwards compatability with the `NearImageArgs` interface. Since it is defined with `image?: string` users can provide `null` or `undefined`. But ideally, it should always be a required field since it is fundamental to the `.with_` method.

So, if it's not provided it gets assigned to `'UNSET'` to be used when performing validation on the `toString()` method. This convo shows that it's convoluted so I will refactor to it to make it simpler 😁